images/flux/main.tf

```diff
@@ -90,3 +90,15 @@ module "tagger" {
     { for t in toset(concat(["latest"], module.version-tags[each.key].tag_list)) : "${t}-dev" => module.latest-dev[each.key].image_ref },
   )
 }
+
+# The slightly more complicated output form that leverages the component's
+# for_each to produce a map of each images required attributes.
+output "images" {
+  value = {
+    for k, v in module.latest : k => {
+      oci_ref     = v.image_ref
+      oci_tags    = module.tagger[k].oci_tags
+      apko_config = v.config
+    }
+  }
+}
```

Comment:
This is where things get hairy and why these aren't autogenerated. This is a dynamic traversal that rolls all the way up through builtin terraform commands (`merge`), that require the terraform runtime, rather than the simple `hcl` parser functions we leverage.

Examples like these are ultimately why we just let the user create this output format, similar to how we already make the user create the tags.